### PR TITLE
Refactor to account for tabstop=8

### PR DESCRIPTION
**Pull request by sirspinach**
rebased on master, reordering commits so "format: Rewrite our format checker in go" comes last.

### CODE REVIEW

**Comment by sirspinach** [comment]
That's strange... the same tests pass on my laptop. Can someone check this out and trying running `make check` on their computer?

`git fetch upstream/pull/472/head:shwang_format && git checkout shwang_format`

**Comment by ejj** [comment]
The second patch introduces trailing whitespace in stitch_test.go (possibly elsewhere as well).

File: scripts/format.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package main
```
Comment (ejj) [inline scripts/format.go]: ... Should this be in this patch?  Did you read the diff before submitting the PR?

**Comment by ejj** [comment]
Btw, I wouldn't bother including the third commit in this PR.  I can just merge it myself once the first two are in.

File: scripts/format.go (hunk 1 above)
Comment (sirspinach) [inline scripts/format.go]: sorry about that :P

**Comment by sirspinach** [comment]
whitespace and third commit removed. Ready for further review.

**Pull request merged**